File: Draft_2/app/migrate_add_tasks_json.py
import sqlite3

def migrate():
    import os
    db_path = os.path.join(os.path.dirname(__file__), "auth.db")
    conn = sqlite3.connect(db_path)
    cur = conn.cursor()
    # Check if tasks_json column exists
    cur.execute("PRAGMA table_info(projects)")
    columns = [row[1] for row in cur.fetchall()]
    if "tasks_json" not in columns:
        cur.execute("ALTER TABLE projects ADD COLUMN tasks_json TEXT")
        print("Added tasks_json column to projects table.")
    else:
        print("tasks_json column already exists.")
    conn.commit()
    conn.close()

if __name__ == "__main__":
    migrate()

File: Draft_1/backend/migrate_add_tasks_json.js
// Migration script to add tasks_json column to projects table (idempotent, PostgreSQL)
const { Pool } = require('pg');

const pool = new Pool({
  connectionString: process.env.DATABASE_URL || 'postgres://postgres:postgres@localhost:5432/uor_app'
});

async function migrate() {
  try {
    // Check if the column already exists
    const checkRes = await pool.query(`
      SELECT column_name
      FROM information_schema.columns
      WHERE table_name='projects' AND column_name='tasks_json'
    `);

    if (checkRes.rows.length === 0) {
      await pool.query('ALTER TABLE projects ADD COLUMN tasks_json TEXT');
      console.log('tasks_json column added.');
    } else {
      console.log('tasks_json column already exists.');
    }
  } catch (err) {
    console.error('Migration failed:', err);
    process.exit(1);
  } finally {
    await pool.end();
  }
}

migrate();